pkg/hvs/tasks/create_default_template.go

```diff
@@ -0,0 +1,464 @@
+/*
+ * Copyright (C) 2020 Intel Corporation
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+package tasks
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/constants"
+	"github.com/intel-secl/intel-secl/v3/pkg/model/hvs"
+
+	"github.com/intel-secl/intel-secl/v3/pkg/hvs/postgres"
+	commConfig "github.com/intel-secl/intel-secl/v3/pkg/lib/common/config"
+	"github.com/intel-secl/intel-secl/v3/pkg/lib/common/setup"
+	"github.com/pkg/errors"
+)
+
+type CreateDefaultTemplate struct {
+	DBConf  commConfig.DBConfig
+	deleted []string
+
+	commandName   string
+	TemplateStore *postgres.FlavorTemplateStore
+}
+
+var defaultFlavorTemplateNames = []string{
+	"default-linux-tpm20-tboot",
+	"default-linux-tpm20-suefi",
+	"default-linux-tpm20-cbnt",
+	"default-uefi",
+	"default-bmc",
+	"default-pfr",
+}
+
+// var defaultFlavorTemplateNames = []string{
+// 	"default-bmc",
+// }
+
+func (t *CreateDefaultTemplate) Run() error {
+	var templates []hvs.FlavorTemplate
+
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize flavor template store instance")
+	}
+
+	if len(t.deleted) != 0 {
+		// Recover delete default template.
+		err := ftStore.Recover(t.deleted)
+		if err != nil {
+			return errors.Wrapf(err, "failed to recover default flavor template(s) %s ", t.deleted)
+		}
+		t.deleted = []string{}
+		return nil
+	}
+
+	templates, err = getTemplates()
+	if err != nil {
+		return err
+	}
+
+	for _, ft := range templates {
+		// create default flavortemplates ONLY if it does not exist already
+		_, err := ftStore.Create(&ft)
+		if err != nil {
+			return errors.Wrap(err, "failed to create default flavor template with ID \""+ft.ID.String()+"\"")
+		}
+	}
+
+	return nil
+}
+
+func (t *CreateDefaultTemplate) Validate() error {
+	ftStore, err := t.flavorTemplateStore()
+	if err != nil {
+		return errors.Wrap(err, "failed to initialize flavor template store instance")
+	}
+	var ftList []hvs.FlavorTemplate
+	defaultFlavorTemplateMap := map[string]bool{}
+	t.deleted = []string{}
+
+	for _, templateName := range defaultFlavorTemplateNames {
+		defaultFlavorTemplateMap[templateName] = false
+	}
+
+	ftList, err = ftStore.Search(false)
```

Comment:
this is not an effective search, search by template names would be a better implementation. The implemented method would have a performance impact.

Reply:
since search is implemented with only whether to include deleted templates or not, this implemented in this way. Once kent provided comment on this, we can modify with name search.